### PR TITLE
Separated install modes for readability, Fixes #49

### LINKED ISSUE #49: Separate install modes from the Operator Install function for readability

### PR DESCRIPTION
**Pull request by makon57**
Separated create group by install mode type into a function, made cleanup function, and added struct for operatorData, not sure if this was the direction intended for readability, and didn't test yet because I don't know how to yet and just wanted to see if this was the right direction or not. Followed preflight code as closely as I understood.

Signed-off-by: Manna Kong <mkong@redhat.com>

### CODE REVIEW

**Comment by madorn** [comment]
Hi @makon57, thanks so much for submitting your PR to `opcap`!

I was able to get this to build and run - looks good!

You did remove the installmode logic out of the `OperatorInstall` func and into `createGroupByInstallMode`.

From the looks of #49, I believe this does reflect the goal of @acmenezes' request...not completely sure. Alex is out till next week but @yashoza19 any thoughts?

File: internal/capability/operator_install.go [hunk 1]
```diff
@@ -49,77 +57,92 @@ func OperatorInstall(s operator.SubscriptionData, c operator.Client) error {
...
-	// TODO: check the name standard for installPlan
-	err = c.InstallPlanApprove(namespace)
-	if err != nil {
-		logger.Debugf("Error creating subscriptions: %w", err)
-		return err
-	}
+func cleanUp(s operator.SubscriptionData, c operator.Client, m operatorData) {
 
-	csvStatus, err := c.WaitForCsvOnNamespace(namespace)
+	csvStatus, err := c.WaitForCsvOnNamespace(m.namespace)
```
Comment (yashoza19) [inline internal/capability/operator_install.go:145]: Just a small nit, this is not part of the clean-up process and can go back to the OperatorInstallAllFromCatalog function.

File: internal/capability/operator_install.go [hunk 2]
```diff
@@ -49,77 +57,92 @@ func OperatorInstall(s operator.SubscriptionData, c operator.Client) error {
...
 
 	if err != nil {
 		logger.Infow("failed", "package", s.Package, "channel", s.Channel, "installmode", s.InstallModeType)
```
Comment (yashoza19) [inline internal/capability/operator_install.go:148]: same suggestion as above for this part.

**Review by yashoza19 — CHANGES_REQUESTED** [review]
Thanks for submitting the PR and working on this issue. I think this function helps in breaking down the OperatorInstallAllFromCatalog function into small parts. I haven't tested the changes locally but left a few comments. @madorn please let me know if those make sense.

File: internal/capability/operator_install.go (hunk 2 above)
Comment (yashoza19) [inline internal/capability/operator_install.go:148]: same suggestion as above for this part.

File: internal/capability/operator_install.go [hunk 3]
```diff
@@ -128,22 +151,21 @@ func OperatorInstall(s operator.SubscriptionData, c operator.Client) error {
 	}
```
Comment (yashoza19) [inline internal/capability/operator_install.go:151]: same suggestion as above for this part.

**Pull request merged**